internal/graph/check.go

```diff
@@ -722,12 +722,15 @@ func TraitsForArrowRelation(ctx context.Context, reader datastore.Reader, namesp
 }
 
 func queryOptionsForArrowRelation(ctx context.Context, ds datastore.Reader, namespaceName string, relationName string) ([]options.QueryOptionsOption, error) {
+	opts := []options.QueryOptionsOption{
```

Comment:
we should figure out the size of the slice ahead of time. Critical path, less allocations.

Reply:
Just allocated it as size 3, which is the max